iqps/upload/forms.py

```diff
@@ -87,6 +120,25 @@ class Meta:
                           Select Others if not found)'
             }
 
+    
+    def save(self, *args, **kwargs):
+        self.subject = self.subject or self.custom_subject
+        # writting to file if a new subject
+        if self.custom_subject is not "":
+            try:
+                with lock:
+                    with open(FILE_PATH,"w") as f:
+                        data = json.load(f)
+                        code_subjects = data["code_subject"]
+                        code_subjects.append(self.customer)
+                        data = { "code_subject": code_subjects }
+                        json.dump(data, f)
+            except:
+                lock.release()
+            finally:
+                lock.release()
```

Comment:
Finally is always executed whether try encounters an error or not, so it is not required to be kept in except.

Also, in the exception block, we should log the error encountered with a traceback.

Reply:
also logged to info level on new subject being added